File: duolingo_rum/Models/ExerciseResult.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace duolingo_rum.Models;

public partial class ExerciseResult
{
    public int Id { get; set; }

    public Guid SessionId { get; set; }

    public int WordId { get; set; }

    [Column("exercise_type")]  // Добавить этот атрибут!
    public string ExerciseType { get; set; } = "translation";

    [Column("user_answer")]    // Добавить этот атрибут!
    public string? UserAnswer { get; set; }

    [Column("is_correct")]     // Добавить этот атрибут!
    public bool IsCorrect { get; set; }

    [Column("response_time_ms")] // Добавить этот атрибут!
    public int? ResponseTimeMs { get; set; }

    [Column("ai_feedback")]    // Добавить этот атрибут!
    public string? AiFeedback { get; set; }

    [Column("answered_at")]    // Добавить этот атрибут!
    public DateTime? AnsweredAt { get; set; }

    public virtual LearningSession Session { get; set; } = null!;
    public virtual Word Word { get; set; } = null!;
}

File: android-app/app/src/main/java/com/linguaai/app/models/ExerciseResult.kt
package com.linguaai.app.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ExerciseResult(
    val id: Int = 0,
    @SerialName("session_id")
    val sessionId: String = "",
    @SerialName("word_id")
    val wordId: Int = 0,
    @SerialName("exercise_type")
    val exerciseType: String = "translation",
    @SerialName("user_answer")
    val userAnswer: String? = null,
    @SerialName("is_correct")
    val isCorrect: Boolean = false,
    @SerialName("response_time_ms")
    val responseTimeMs: Int? = null,
    @SerialName("ai_feedback")
    val aiFeedback: String? = null,
    @SerialName("answered_at")
    val answeredAt: String? = null
)
